haskell/test/SMT.hs

```diff
@@ -660,18 +668,21 @@ applyBurn ccSender cs burnparams = do
   void $ ensureMinter ccSender cs
   let totalBurn = sum burnparams
   if totalBurn <= (fromMaybe 0 $ Map.lookup ccSender $ ssLedger cs)
-    then let
-      burn value storage =
-        storage { ssLedger = Map.alter (\case
-          Nothing -> if value > 0
-            then error ("Unexpected burn:" <> show value)
-            else Nothing
-                -- As per FA2, burn should follow the tranfer logic, which allows zero transfer
-                -- from non-existant accounts. So here we should do nothing, instead of throwing an
-                -- error.
-          Just x -> let b = x - value
-            in if b > 0 then Just b else Nothing) ccSender $ ssLedger cs }
-      in Right $ foldr burn cs burnparams
+    then if totalBurn > ssTotalSupply cs
+      then Left NEGATIVE_TOTAL_SUPPLY
```

Comment:
AFAIU if our test generator is written correctly, `NEGATIVE_TOTAL_SUPPLY` should never occur. Isn't it better to use `error` here to indicate that there is a bug in the test itself? For example, few lines below we are doing `error ("Unexpected burn:" <> show value)` when we try to burn non-existing tokens.

Reply:
True.